Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -118,7 +118,16 @@ def amiPhysValues(data):
     - k_factor
     - cross_section_ref
     """
-    dataset = data['datasetname']
+    if not isinstance(data, dict):
+        sys.stderr.write("(WARN) Cannot update non-dict data:%r\n" % data)
+        return False
+    dataset = data.get('datasetname')
+    if dataset is None:
+        sys.stderr.write("(WARN) No datasetname found in data:%r\n" % data)
+        return False
+    elif not isinstance(dataset, str) and not isinstance(dataset, unicode):
```

Comment:
*Just a comment:* `isinstance()` can work with a number of types/classes, and returns `True` if at least one class matches:
```
elif not isinstance(dataset, (str, unicode)):
    ...
```

Reply:
Hm. If my memory serves, I looked for something like this, and found nothing, which means I've searched poorly. Thanks for enlightening me. Updated.